src/lightning_lite/accelerators/mps.py

```diff
@@ -40,15 +39,16 @@ def teardown(self) -> None:
     @staticmethod
     def parse_devices(devices: Union[int, str, List[int]]) -> Optional[List[int]]:
         """Accelerator device parsing logic."""
-        parsed_devices = device_parser.parse_gpu_ids(devices, include_mps=True)
+        from lightning_lite.utilities.device_parser import parse_gpu_ids
```

Comment:
should we maybe also split this into mps and cuda specific logic? Currently most of the relevant functions have logic separated by if-else blocks anyways.

Reply:
I would prefer that. Semantically, it is also looks strange that we add up the list of indices for gpus and mps gpus: (https://github.com/Lightning-AI/lightning/pull/13123#discussion_r884292772)

Reply:
Yes. I said the same in https://github.com/Lightning-AI/lightning/pull/14753#discussion_r973117360